### PR TITLE
Kans/session

### PR DESCRIPTION
**Pull request by kans**
## Summary by CodeRabbit

- **New Features**
  - Session caching: in-memory and gRPC-backed caches with context-bound sync IDs and JSON helpers.
  - Automatic session cache wiring in CLI, Lambda, and server flows; Lambda uses DPoP-authenticated gRPC client.
  - Sync ID propagation added to connector requests and server contexts; new session-cache service with validation-enabled APIs.

- **Documentation**
  - Added session cache README describing implementations and usage.

- **Tests**
  - Extensive tests for memory and gRPC caches, client creation, pagination, concurrency, and edge cases.

I tested this manually locally and in a lambda function.  Syncs are ok in both.

### CODE REVIEW

**Review by ggreer — APPROVED** [review]

File: pkg/session/grpc_session.go [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+	// Create TLS transport credentials
+	systemCertPool, err := x509.SystemCertPool()
+	if err != nil || systemCertPool == nil {
+		return nil, fmt.Errorf("failed to load system cert pool: %w", err)
+	}
+	transportCreds := credentials.NewTLS(&tls.Config{
+		RootCAs:    systemCertPool,
+		MinVersion: tls.VersionTLS12,
+	})
+
```
[the remark this answers is not in the document]
Comment (kans) [reply pkg/session/grpc_session.go:94]: This has the lambda tag on it - it will only run in a container in AWS.

File: pkg/session/grpc_session.go [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+	}
+
+	// Create the gRPC connection
+	conn, err := grpc.DialContext(ctx, addr, dialOpts...) //nolint:staticcheck // grpc.DialContext is deprecated but we are using it still.
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to session service at %s: %w", addr, err)
+	}
+
+	return v1.NewBatonSessionServiceClient(conn), nil
+}
```
[the remark this answers is not in the document]
Comment (kans) [reply pkg/session/grpc_session.go:109]: This is only for lambda functions.
